src/content/learning/how-durable-objects-work.md

```diff
@@ -0,0 +1,216 @@
+        // TODO: txn example
+    }
+
+}
+
+```
+
+Transactions can fail if they conflict with a concurrent transaction from the same Durable Object.  Transactions are transparently and automatically retried once before returning a failure.
+
+
+## Binding the class to a Worker
+
+As with Workers KV, the class namespace must be created and then bound into a Worker before it can be used.
+
+First, upload the script that contains the class via the Workers Dashboard or via the command line.
+
+```sh
+curl -i -H "X-Auth-Email: ${EMAIL}" -H "X-Auth-Key: ${AUTH_KEY}" "https://api.cloudflare.com/client/v4/accounts/${ACCOUNT_TAG}/workers/scripts/${SCRIPT_NAME}" -X PUT -H "Content-Type: application/javascript" --data "`cat durable_object_example.js`"
+```
+
+To create the class, call the following endpoint.
+
+```sh
+curl -i -H "X-Auth-Email: ${EMAIL}" -H "X-Auth-Key: ${AUTH_KEY}" "https://api.cloudflare.com/client/v4/accounts/${ACCOUNT_TAG}/workers/durable_objects/namespaces" -X POST --data "{'name': 'example-class', 'script': ${SCRIPT_NAME}"
+```
+
+You can now create a binding for the Worker which uses this class.  TODO: Not sure if this is still the correct syntax, check.
+
+```js 
+// binding.json
+{
+  "body_part": "script",
+  "bindings": [
+    {
+      "type": "durable_object_namespace",
+      "name": "example-class",
+      "namespace_id": "<namespace id>"
+    }
+  ]
+}
+
+```
+
+
+## Instantiating a Durable Object
+
+Creating a Durable Object requires retrieving a Fetcher object from the class in your Workers Script.  Once you have the Fetcher object, you can use it to send [Requests](/runtime-apis/request) to the Durable Object instance.
+
+```js
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+ 
+async function handleRequest(request) {
+  // Retrieve the Fetcher for the Durable Object with id 1
+  let durableObject = await DURABLE_OBJECT_CLASS.get("1")
+  // Send a request to Durable Object '1', with the URI "get-count"
+  let resp = await durableObject.fetch("get-count")
+  let count = await resp.text()
+ 
+  // Regardless of the URI, requests are always routed to the Durable Object
+  resp = await durableObject.fetch()
+  let message = await resp.text()
+
+  // TODO: system generated id example
+  let id = generateId()
+  let sysDurableObject = await DURABLE_OBJECT_CLASS.get()
+  resp = await sysDurableObject.fetch('get-count')
+  let sysCount = await resp.text()
+ 
+  return new Response("Object '1' count: " + count + " and message " + message + ".  System Object '" + id + "' count: " + sysCount)
+}
+
+```
+
+Learn more at the [Workers Durable Objects API reference](/runtime-apis/durable-objects).
+
+## Limitations and considerations when working with Durable Objects
+
+### Global Uniqueness
+
+Uniqueness is currently enforced only when a Durable Object begins execution.  In the future, uniqueness will also be enforced whenever a Durable Object makes storage requests.  
```

Comment:
I'd explicitly call out that this means when a script is updated, existing requests may still be running on the old object concurrently with new requests going to the new object.